src/korg_nanokontrol/korg.py

```diff
@@ -0,0 +1,158 @@
+#!/usr/bin/env python
+#
+# joystick input driver for Korg NanoKontrol input device
+#
+# Author: Austin Hendrix
+# Author: Allison Thackston
+
+import rospy
+import rospkg
+import rosparam
+
+import pygame
+import pygame.midi
+
+from sensor_msgs.msg import Joy
+
+
+class KorgNanoKontrol(object):
+
+    ''' Class to connect to and publish the Korg NanoKontrol midi device
+        as a ros joystick
+    '''
+
+    def __init__(self):
+        ''' connect to midi device and set up ros
+        '''
+        pygame.midi.init()
+        devices = pygame.midi.get_count()
+        if devices < 1:
+            rospy.logerr("No MIDI devices detected")
+            exit(-1)
+        rospy.loginfo("Found %d MIDI devices" % devices)
+
+        input_dev = int(rospy.get_param(
+            "input_dev", pygame.midi.get_default_input_id))
+
+        rospy.loginfo("Using input device %d" % input_dev)
+
+        self.controller = pygame.midi.Input(input_dev)
+
+        # load in default parameters if not set
+        if not rospy.has_param('modes'):
+            rospack = rospkg.RosPack()
+            paramlist = rosparam.load_file(
+                rospack.get_path('korg_nanokontrol')+'/config/nano_kontrol_config.yaml')
+            for params, ns in paramlist:
+                rosparam.upload_params(ns, params)
+
+        self.modes = rospy.get_param('modes')
+
+        # determine how axis should be interpreted
+        self.center_axis = rospy.get_param('center_axis', True)
+
+        self.msg = Joy()
+        self.mode = None
+
+        self.pub = rospy.Publisher('joy', Joy, latch=True)
```

Comment:
I know it was missing this before your PR, but could you add a `queue_size` parameter here? @ahendrix what is a good value for it?